src/execute/julia.ts

```diff
@@ -0,0 +1,455 @@
+import { error, info } from "log/mod.ts";
+import { join } from "path/mod.ts";
+import { MappedString, mappedStringFromFile } from "../core/mapped-text.ts";
+import { partitionMarkdown } from "../core/pandoc/pandoc-partition.ts";
+import { readYamlFromMarkdown } from "../core/yaml.ts";
+import { ProjectContext } from "../project/types.ts";
+import {
+  DependenciesOptions,
+  ExecuteOptions,
+  ExecuteResult,
+  ExecutionEngine,
+  ExecutionTarget,
+  kJuliaEngine,
+  PandocIncludes,
+  PostProcessOptions,
+} from "./types.ts";
+import { jupyterAssets, jupyterToMarkdown } from "../core/jupyter/jupyter.ts";
+import {
+  kExecuteDaemon,
+  kExecuteDaemonRestart,
+  kExecuteDebug,
+  kFigDpi,
+  kFigFormat,
+  kFigPos,
+  kIpynbProduceSourceNotebook,
+  kKeepHidden,
+} from "../config/constants.ts";
+import {
+  isHtmlCompatible,
+  isIpynbOutput,
+  isLatexOutput,
+  isMarkdownOutput,
+  isPresentationOutput,
+} from "../config/format.ts";
+import { resourcePath } from "../core/resources.ts";
+import { quartoRuntimeDir } from "../core/appdirs.ts";
+import { normalizePath } from "../core/path.ts";
+import { isInteractiveSession } from "../core/platform.ts";
+import { runningInCI } from "../core/ci-info.ts";
+import { sleep } from "../core/async.ts";
+import { JupyterNotebook } from "../core/jupyter/types.ts";
+import { existsSync } from "fs/mod.ts";
+
+export interface JuliaExecuteOptions extends ExecuteOptions {
+  julia_cmd: string;
+  oneShot: boolean; // if true, the file's worker process is closed before and after running
+  supervisor_pid?: number;
+}
+
+export const juliaEngine: ExecutionEngine = {
+  name: kJuliaEngine,
+
+  defaultExt: ".qmd",
+
+  defaultYaml: () => [],
+
+  defaultContent: () => [
+    "```{julia}",
+    "1 + 1",
+    "```",
+  ],
+
+  validExtensions: () => [],
+
+  claimsFile: (file: string, ext: string) => false,
+
+  claimsLanguage: (language: string) => {
+    return language.toLowerCase() === "julia";
+  },
+
+  partitionedMarkdown: async (file: string) => {
+    return partitionMarkdown(Deno.readTextFileSync(file));
+  },
+
+  // TODO: ask dragonstyle what to do here
+  executeTargetSkipped: () => false,
```

Comment:
In some code paths, Quarto needs to create an engine and `ExecutionTarget` even if we don't intend to eventually execute the markdown.

In those cases, we call `executeTargetSkipped()` to signal that the specific target was skipped. In the Jupyter engine, this causes a cleanup of the notebook. If you don't create transient files prior to the execution, then I think this is fine.